### PR TITLE
Add Key.SetCert() function

### PR DESCRIPTION
**Pull request by jessieqliu**
Add a SetCert() function to client.Key that allows setting the AK Certificate. The function will verify that the cert matches Key.pubKey.

### CODE REVIEW

**Review by jkl73 — APPROVED** [review]
LGTM

File: client/keys.go [hunk 1]
```diff
@@ -468,6 +470,21 @@ func (k *Key) CertDERBytes() []byte {
...
 }
 
+// SetCert assigns the provided certificate to the key after verifying it matches the key.
+func (k *Key) SetCert(cert *x509.Certificate) error {
+	certPubKey, ok := cert.PublicKey.(crypto.PublicKey)
+	if !ok {
+		return errors.New("certificate public key is not the appropriate type")
+	}
+
+	if !reflect.DeepEqual(certPubKey, k.pubKey) {
```
Comment (josephlr) [inline client/keys.go:480]: I think we already have a method to compare two `crypto.PublicKey`s

https://github.com/google/go-tpm-tools/blob/66ec981c39007db9718f14f34c15102cba642499/server/verify.go#L125

We should move that method from `server` to `internal` and then just use it here

File: client/keys.go [hunk 2]
```diff
@@ -468,6 +470,21 @@ func (k *Key) CertDERBytes() []byte {
 	return k.cert.Raw
 }
 
+// SetCert assigns the provided certificate to the key after verifying it matches the key.
+func (k *Key) SetCert(cert *x509.Certificate) error {
```
Comment (josephlr) [inline client/keys.go:474]: Nit: we should also be using this method internally for the various methods that set certificates, basically the 4 methods that call: `getCertificateFromNvram`.

Comment (josephlr) [reply client/keys.go:474]: You may have to change that helper function to be a helper method, but it should be fine.

**Comment by josephlr** [comment]
@jkl73 @jessieqliu I added two commits to Jessie's changes. Can you take a look (specifically at https://github.com/google/go-tpm-tools/pull/167/commits/11161ba2d64d6289c68c0d2cc6bef86b36ff85ed). If my changes are OK, merge this PR.

**Review by josephlr — APPROVED** [review]

**Review by jkl73 — APPROVED** [review]
lgtm

**Pull request merged**